client/my-sites/media-library/upload-controls.jsx

```diff
@@ -0,0 +1,186 @@
+import MediaUtils from 'lib/media/utils';
+import SectionHeader from 'components/section-header';
+import { userCan } from 'lib/site/utils';
+import { VideoPressFileTypes } from 'lib/media/constants';
+
+class MediaLibraryUploadControls extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			addingViaUrl: false,
+		};
+	}
+
+	onAddUrlClick = () => {
+		! this.state.addingViaUrl && this.setState( { addingViaUrl: true } );
+	};
+
+	onAddUrlClose = () => {
+		this.setState( { addingViaUrl: false } );
+	};
+
+	uploadFiles = ( event ) => {
+		if ( event.target.files && this.props.site ) {
+			MediaActions.clearValidationErrors( this.props.site.ID );
+			MediaActions.add( this.props.site.ID, event.target.files );
+		}
+
+		ReactDom.findDOMNode( this.refs.itemForm ).reset();
+		ReactDom.findDOMNode( this.refs.dropzoneForm ).reset();
+		this.props.onAddMedia();
+		analytics.mc.bumpStat( 'editor_upload_via', 'add_button' );
+	};
+
+	onUploadClick = () => {
+		if ( this.props.href ) {
+			page( this.props.href );
+		}
+	};
+
+	onExternalServiceClick = () => {
+		this.props.onSourceChange( 'google_photos' );
+	};
+
+	/**
+	 * Returns a string of comma-separated file extensions supported for the
+	 * current site, to be used as the `accept` attribute in an `input` element
+	 * of type `file`. This is a non-standard use of the `accept` attribute,
+	 * but is supported in Internet Explorer and Chrome browsers. Further input
+	 * validation will occur when attempting to upload the file.
+	 *
+	 * @return {string} Supported file extensions, as comma-separated string
+	 */
+	getInputAccept = () => {
+		if ( ! MediaUtils.isSiteAllowedFileTypesToBeTrusted( this.props.site ) ) {
+			return null;
+		}
+		const allowedFileTypesForSite = MediaUtils.getAllowedFileTypesForSite( this.props.site );
+
+		return uniq( allowedFileTypesForSite.concat( VideoPressFileTypes ) ).map( ( type ) => `.${ type }` ).join();
+	};
+
+	renderUploadViaUrl() {
+		const { site, translate, onAddMedia } = this.props;
+
+		if ( this.state.addingViaUrl ) {
+			return (
+				<CompactCard className="media-library__upload-controls-option">
+					<MediaLibraryUploadUrl
+						site={ site }
+						onAddMedia={ onAddMedia }
+						onClose={ this.onAddUrlClose } />
+				</CompactCard>
+			);
+		}
+
+		return (
+			<CompactCard className="media-library__upload-controls-option" onClick={ this.onAddUrlClick }>
+				<Gridicon icon="domains" size={ 18 } />
+				<span className="media-library__upload-controls-option-name">{ translate( 'URL' ) }</span>
+				<Gridicon icon="chevron-right" />
+			</CompactCard>
+		);
+	}
+
+	renderFileItem() {
+		const { translate } = this.props;
+
+		return (
+			<form ref="itemForm">
+				<CompactCard className="media-library__upload-controls-option">
+					<Gridicon icon="computer" size={ 18 } />
+					<span className="media-library__upload-controls-option-name">{ translate( 'Computer' ) }</span>
```

Comment:
Hi! I've found a possible matching string that has already been translated **96** times:
`translate( 'From Computer' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).